mlir/include/mlir/Interfaces/LoopLikeInterface.h

```diff
@@ -57,4 +57,24 @@ class HasParallelRegion : public TraitBase<ConcreteType, HasParallelRegion> {
 /// Include the generated interface declarations.
 #include "mlir/Interfaces/LoopLikeInterface.h.inc"
 
+namespace mlir {
+/// A function that rewrites `target`'s terminator as a teminator obtained by
+/// fusing `source` into `target`.
+using FuseTerminatorFn =
+    std::function<void(RewriterBase &rewriter, LoopLikeOpInterface source,
```

Comment:
If the function reference doesn't need to outlive the function it is passed into, use `function_ref`. This avoids a dynamic allocation.

Reply:
ok. although that is the case now, i'm not sure if that will always be the case.  I figured its usage would be similar to `NewYieldValuesFn`, but will defer to reviewers on this.

Reply:
When somebody wants to change that, it's up to them to update the typedef. We shouldn't be paying the cost for unlikely hypotheticals.  I suspect we should update `NewYieldValuesFn` to be a `function_ref` as well. There is a lot of suboptimal data structure use in loop/tiling interfaces unfortunately.

Reply:
that's fair. thanks. will change

Reply:
i got crashes when changing to `function_ref` for `NewYieldValuesFn`, but didn't really look into it so just keeping as is